### ISSUE TITLE
Replacing a lot of entries is still extremely slow compared to other editors

### ISSUE THREAD

**Issue by maximeborges** [zed-industries/zed#14408]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

When trying to replace a lot of tabs by semicolons in a CSV-like file, Zed basically froze for a few minutes (I killed it after 3 minutes).
I was on 0.139.3, I checked the existing issues and saw that https://github.com/zed-industries/zed/issues/13455 (and even the file example from https://github.com/zed-industries/zed/issues/11210 takes tens of seconds just to load up) were fixed in 0.143.6, so I tried installing zed-preview-bin from AUR which currently is 0.144.2, but I don't really see much of an improvement.

## Repro
You can generate a file with similar content to my test file with (210k matches):
``` bash
for i in {1..30000}; do echo "$i\t$i\t$i\t$i\t$i\t$i\t$i\t$i" >> /tmp/test_file; done
```
Then in Zed, use the replace function with a tab char (either by pasting a tab char in the search field, or enabling regexp and using `\t`; both takes the same time on my machine) and use a semicolon in the replace field.
This action takes about 45s, using 100% of a thread of my i7-1260p CPU.
The same action takes less than half a second in VSCode, and is instantaneous in vim.

### Environment

Zed: v0.144.2 (Zed Preview)
OS: Linux Wayland manjaro unknown
Memory: 31.1 GiB
Architecture: x86_64

**Comment by osiewicz** [comment]
Thanks for reporting this interesting issue =)
For what it's worth, this issue is a bit different than #13455 - in this case, our internal data structures (TabMap) which are responsible for tracking code layout take quite a bit to update since the # of changes is pretty large (210k to be exact).

**Comment by maximeborges** [comment]
I was thinking about some layout issue also, but since I currently have no clue on how displaying things on the screen is done in Zed, I thought maybe it was still something related to this previous issue.
Good to know it's something different and I'm not badly testing the new release ^^

**Comment by osiewicz** [comment]
No, this is a very valid issue (and for the record, I can reproduce it just fine). Manual CSV processing is not unheard of, so I'll look into it. Thanks :)

**Comment by osiewicz** [comment]
The initial fix for it landed in https://github.com/zed-industries/zed/pull/14567, which will be released to Preview channel tomorrow. For me it slashed the time needed for the repro to complete from 13s to 2s, so if in your case I'd expect it to not require killing off the app at the very least. :)
I'm gonna leave this issue open though, as there's still more that we can do.

**Comment by phoenisx** [comment]
I have the following system specs:
```
OS: Pop!_OS 22.04 LTS x86_64
Kernel: 6.9.3-76060903-generic
CPU: AMD Ryzen 5 5625U with Radeon Graphics (12) @ 4.388GHz
GPU: AMD ATI 04:00.0 Barcelo
Memory: 12946MiB / 14805MiB
```
and my DM/WM (not sure) got stuck for a few minutes when I tried to do a simple regex search, e.g. `on\(('|")ready('|")`

Not sure if this is related, but I felt like sharing as I had to switch off my laptop for a few minutes and then everything worked fine.

**Comment by notpeter** [comment]
Just following up and confirming this is still an issue with the current Zed.
With your test case VSCode is instantaneous (<300ms) and in Zed it is noticeably slower (>2secs).

**Comment by maximeborges** [comment]
This seems to still be an issue, the current test setup takes about 3s on my Ryzen 5950x and is almost instantaneous on VScode.
Testing with 0.214.4
